### ISSUE TITLE
Migrate from opentracing to opentelemetry

### ISSUE THREAD

**Issue by wbarnha** [faust-streaming/faust#382]
The `opentracing` project is deprecated in favor of `opentelemetry`. Due to ongoing flaws with `opentracing`, we have had difficulty troubleshooting particular bugs that rely on the usage of `opentracing.Span` objects.

We will need to miigrate `opentracing` usage over to `opentelemetry` as documented in https://opentelemetry-python.readthedocs.io/en/stable/shim/opentracing_shim/opentracing_shim.html.

**Comment by srikanthccv** [comment]
👋  (I am one of the maintainers of OpenTelemetry). 
>We will need to miigrate opentracing usage over to opentelemetry as documented in https://opentelemetry-python.readthedocs.io/en/stable/shim/opentracing_shim/opentracing_shim.html.

Shim is aimed at users who want to use OTLP protocol while the code is still instrumented with OpenTracing and plan for incremental migration in large codebases. I looked at `opentracing` usage Faust codebase. I think you are better off truly migrating over to OpenTelemetry by directly instrumenting with `opentelemetry-api`. Let us know if you need any clarification and help with this.

**Comment by lqhuang** [comment]
Actually, one of my targets for `mode-ng` is to integrate OpenTelemetry protocol. Of course, long way to go 😭

**Comment by Chiraguthra** [comment]
> 👋 (I am one of the maintainers of OpenTelemetry).
> 
> > We will need to miigrate opentracing usage over to opentelemetry as documented in https://opentelemetry-python.readthedocs.io/en/stable/shim/opentracing_shim/opentracing_shim.html.
> 
> Shim is aimed at users who want to use OTLP protocol while the code is still instrumented with OpenTracing and plan for incremental migration in large codebases. I looked at `opentracing` usage Faust codebase. I think you are better off truly migrating over to OpenTelemetry by directly instrumenting with `opentelemetry-api`. Let us know if you need any clarification and help with this.

Hi @srikanthccv .. if we start using opentelemetry that would be manual instrumentation.. right? since currently it doesnt support automatic instrumentation for faust. My question in other thread was, do we have any method available which can be used as decorator over to span child traces for the same. any lead on this?

**Comment by mknaw** [comment]
Seems additionally pressing given `opentracing` `setup.cfg` noncompliance with [the new expectations of setuptools 78.0.1](https://setuptools.pypa.io/en/stable/history.html#v78-0-0)

**Comment by dada-engineer** [comment]
yep looking into it... @patkivikram do you have experience with opentelemetry? There also is an aiokafka instrumentation, we might be able to utilize.

@srikanthccv what is the best resource to get startet for someone that has not yet used OpenTelemetry?
